### DISCUSSION TITLE
File not reinstalling during repair due to uncompressed

### DISCUSSION THREAD

**Discussion by kalyani1217** [wixtoolset/issues#8666] [Questions] [answered]
I have created an Addon Setup by setting RelatedBundle Action to Addon.

I have installed the base setup and then the Addon setup.

Now,  delete any file, and repair the base setup. 

Both the base setup and the add-on setup are being repaired. However, the file is of base setup not of addon.

the addon log says: Source for file 'ComponentCC2.txt' is uncompressed, at 'C:\windows\Installer\$PatchCache$\Managed\7CE287722CD1031409FD983276435449\1.0.1\'.

```
Executing op: SetTargetFolder(Folder=C:\Program Files (x86)\SmartPlant\Custom1\FeatureC\FeatureC2\)
MSI (s) (50:9C) [16:09:55:470]: Executing op: FileCopy(SourceName=ComponentCC2.txt,SourceCabKey=ComponentCC2.txt,DestName=ComponentCC2.txt,Attributes=512,FileSize=2080033,PerTick=65536,,VerifyMedia=0,,,,,CheckCRC=0,,,InstallMode=63438848,HashOptions=0,HashPart1=1510649027,HashPart2=-975706500,HashPart3=-666771121,HashPart4=-2062273328,,)
MSI (s) (50:9C) [16:09:55:478]: File: C:\Program Files (x86)\SmartPlant\Custom1\FeatureC\FeatureC2\ComponentCC2.txt;	Overwrite;	Won't patch;	REINSTALLMODE specifies all files to be overwritten
MSI (s) (50:9C) [16:09:55:478]: Source for file 'ComponentCC2.txt' is uncompressed, at 'C:\windows\Installer\$PatchCache$\Managed\7CE287722CD1031409FD983276435449\1.0.1\'.
MSI (s) (50:9C) [16:09:55:482]: Re-applying security from existing file.
MSI (s) (50:9C) [16:09:55:501]: Verifying accessibility of file: ComponentCC2.txt
MSI (s) (50:9C) [16:09:55:510]: Note: 1: 2318 2:  
MSI (s) (50:9C) [16:09:55:518]: Note: 1: 2318 2:  
MSI (s) (50:9C) [16:09:55:530]: Note: 1: 2318 2: C:\Program Files (x86)\SmartPlant\Custom1\FeatureC\FeatureC2\ComponentCC2.txt 
MSI (s) (50:9C) [16:09:55:530]: File will have security applied from OpCode.
```

Why the file is not uncompressed? I have added a media template as well.
    `<Media  Id="1"  Cabinet="Data1.cab"  EmbedCab="yes" VolumeLabel="DISK1" />`

I would be grateful if you could help me out.

**Comment by bevanweiss** [comment, marked as answer]
I'm a bit confused by what you have here.
It sounds like you have a Bundle A, which contains at least one MSI B
And then you have another Bundle C, which must also contain at least one MSI D.
You say that C has a RelatedBundle attribute that 'points' back to Bundle A (as an Addon), but you've made no comment about the MSIs.
If MSI D is an Upgrade of MSI B, do you have this relationship authored in MSI D?

I'd think if you've got Bundle C as an Addon to Bundle A, and not as a Patch.. then it really shouldn't be changing files deployed as part of Bundle A.
Does this shared file have consistent component IDs between the different MSIs?
and is the keypath the same between them?

**Reply by kalyani1217** [reply]
Thanks @bevanweiss for the quick response.

After adding KeyPath to the component instead of to File in both base and addon MSIs, the addon files are reinstalled when I repair the base setup(both base and addon setups are repaired).
However, in my base msi, KeyPath was added to File and it was already released. I cannot add the KeyPath to the component now in the base msi as it was already released.

Adding KeyPath to the component, only in the addon msi does not work.

Any workaround for this issue is greatly appreciated.

Thanks,
Kalyani

**Reply by bevanweiss** [reply]
The general rule is that each file that is "important enough", it should be its own component, as its own keypath.
Unfortunately if you've broken that, it'll be a bit tough to resolve it.
You could deploy an upgrade bundle, with an upgrade MSI (to the old MSI).
Then you should get to start fresh on those things... since all the old would be removed, and the new should take its place.

**Reply by kalyani1217** [reply]
Got it. Thank you @bevanweiss
